### PR TITLE
Fix rule: prefer_initializing_formals

### PR DESCRIPTION
**Pull request by JPaulsen**
Fixes https://github.com/dart-lang/linter/issues/493

### CODE REVIEW

File: lib/src/rules/prefer_initializing_formals.dart [hunk 1]
```diff
@@ -94,16 +97,20 @@ class _Visitor extends SimpleAstVisitor {
   @override
   visitConstructorDeclaration(ConstructorDeclaration node) {
     final parameters = _getParameters(node);
+    final parametersUsedOne = new Set<Element>();
+    final parametersUsedMoreThanOne = new Set<Element>();
```
Comment (bwilkerson) [inline lib/src/rules/prefer_initializing_formals.dart]: For the two lines above, "One" --> "Once".

File: lib/src/rules/prefer_initializing_formals.dart [hunk 2]
```diff
@@ -112,9 +119,30 @@ class _Visitor extends SimpleAstVisitor {
...
           expression is SimpleIdentifier &&
-          parameters.contains(expression.bestElement);
+          parameters.contains(expression.bestElement) &&
+          (!parametersUsedMoreThanOne.contains(expression.bestElement) ||
+              constructorFieldInitializer.fieldName.bestElement?.name ==
+                  expression.bestElement.name);
     }
 
+    void processElement(Element element) {
+      if (parametersUsedOne.contains(element)) {
```
Comment (bwilkerson) [inline lib/src/rules/prefer_initializing_formals.dart]: With sets, you can do this more efficiently by adding the element to the set and then using the value returned by `add` to determine whether it was already in the set.

File: lib/src/util/dart_type_utilities.dart [hunk 3]
```diff
@@ -18,6 +18,11 @@ class DartTypeUtilities {
       (type is InterfaceType &&
           extendsClass(type.superclass, className, library));
 
+  static Element getRealElement(Element element) =>
+      element is PropertyAccessorElement
+          ? getRealElement(element.variable)
```
Comment (bwilkerson) [inline lib/src/util/dart_type_utilities.dart]: The variable of a PropertyAccessorElement will never be a PropertyAccessorElement, so the recursive call is not necessary.

File: lib/src/rules/prefer_initializing_formals.dart (hunk 1 above)
Comment (JPaulsen) [inline lib/src/rules/prefer_initializing_formals.dart]: Fixed

File: lib/src/rules/prefer_initializing_formals.dart (hunk 2 above)
Comment (JPaulsen) [inline lib/src/rules/prefer_initializing_formals.dart]: Fixed

File: lib/src/util/dart_type_utilities.dart (hunk 3 above)
Comment (JPaulsen) [inline lib/src/util/dart_type_utilities.dart]: Fixed

**Pull request merged**